src/app/Processor/ProcessorEdit/ProcessorEdit.tsx

```diff
@@ -45,8 +52,81 @@ const ProcessorEdit = (props: ProcessorEditProps): JSX.Element => {
     type: "",
     parameters: {},
   });
+  const [request, setRequest] = useState<ProcessorRequestData>({
+    name,
+    type: processorType,
+    filters,
+    transformationTemplate: transformation,
+    action,
+    source,
+  });
+  const [isSubmitted, setIsSubmitted] = useState(false);
 
-  const { t } = useTranslation(["openbridgeTempDictionary"]);
+  const {
+    registerValidateConfig,
+    validateName,
+    validation,
+    validate,
+    resetValidation,
+  } = useValidateProcessor(request, existingProcessorName);
+
+  useEffect(() => {
+    setRequest({
+      name,
+      type: processorType,
+      filters,
+      transformationTemplate: transformation,
+      action,
+      source,
+    });
+  }, [name, processorType, filters, transformation, action, source]);
+
+  const handleSubmit = (): void => {
+    setIsSubmitted(true);
+    if (validate() && request) {
+      const requestData: ProcessorRequest = { name: request.name };
+      if (request.type === "sink") {
+        requestData.action = request.action;
+      } else {
+        requestData.source = request.source;
+      }
+      if (request.filters && request.filters.length > 0) {
+        const filtersData = request.filters.filter(
+          (filter) => filter.type && filter.value && filter.key
+        );
+        if (filtersData.length > 0) {
+          requestData.filters =
+            filtersData as unknown as ProcessorRequest["filters"];
+        }
+      }
+      if (
+        request.transformationTemplate &&
+        request.transformationTemplate.trim().length > 0
+      ) {
+        requestData.transformationTemplate =
+          request.transformationTemplate.trim();
+      }
+      onSave(requestData);
+    }
+  };
```

Comment:
It seems that this function is performing too many actions: why don't you split it by multiple sub-functions?

Reply:
The big chunk here is about a single task, preparing the request data to be sent to the APIs. The data from the form has more information than what's needed.

But yes, that's maybe too much for a `handleSubmit`, I've moved the request preparation to a separate function.